src/main/cpp/src/api/annotation_service.cc

```diff
@@ -48,96 +100,40 @@ AnnotationService::AnnotationService(const std::string& export_configuration) {
       throw GenomicsDBException(logger.format("Annotation data source {} not found", filename));
     }
     genomicsdb_htslib_plugin_initialize(filename.c_str());
-    std::vector<std::string> fields;
+    std::set<std::string> fields;
     for(auto field : export_config.annotation_source(i).attributes()) {
-      fields.push_back(field);
+      fields.insert(field);
     }
-    annotation_source_t source(filename, export_config.annotation_source(i).data_source(), fields);
-    m_annotation_sources.push_back(std::move(source));
+    m_annotation_sources.emplace_back(filename, export_config.annotation_source(i).data_source(), fields);
   }
-  m_annotation_buffer.reserve(32);
+  m_annotation_buffer.reserve(64);
 }
 
 std::vector<annotation_source_t>& AnnotationService::get_annotation_sources() {
   return m_annotation_sources;
 }
 
+std::unordered_map<int, int> bcf_ht_type_to_nbytes = {
```

Comment:
VariantFieldTypeUtil::size(bcf_ht_type) gets you this  (variant_field_data.h)

Reply:
Ah, thanks!